packages/edit-site/src/components/pattern-actions/index.js

```diff
@@ -0,0 +1,281 @@
+	// Only custom patterns or custom template parts can be renamed or deleted.
+	const isUserPattern = record?.type === PATTERN_TYPES.user;
+	const isNonUserPattern = item?.type === PATTERN_TYPES.theme;
+	const isTemplate = record?.type === TEMPLATE_POST_TYPE;
+	const isTemplatePart = record?.type === TEMPLATE_PART_POST_TYPE;
+
+	if (
+		! isTemplatePart &&
+		! isTemplate &&
+		! isUserPattern &&
+		! isNonUserPattern
+	) {
+		return null;
+	}
+
+	const isRemovable = isTemplateRemovable( record );
+	const isRevertable = isTemplateRevertable( record );
+	const isEditable = isUserPattern || isRemovable;
+	// Only patterns and template parts can be duplicated for now.
+	const isDuplicable = isUserPattern || isNonUserPattern || isTemplatePart;
+	const decodedTitle = decodeEntities(
+		record?.title?.rendered || record?.title?.raw
+	);
+
+	const exportAsJSON = ( pattern ) => {
+		const json = {
+			__file: pattern.type,
+			title: pattern?.title?.raw || pattern.slug,
+			content: pattern.content.raw,
+			syncStatus: pattern.wp_pattern_sync_status,
+		};
+
+		return downloadBlob(
+			`${ kebabCase( json.title ) }.json`,
+			JSON.stringify( json, null, 2 ),
+			'application/json'
+		);
+	};
+
+	const deletePattern = async ( pattern ) => {
+		try {
+			await __experimentalDeleteReusableBlock( pattern.id );
+			createSuccessNotice(
+				sprintf(
+					// translators: %s: The pattern's title e.g. 'Call to action'.
+					__( '"%s" deleted.' ),
+					decodedTitle
+				),
+				{ type: 'snackbar', id: 'edit-site-patterns-success' }
+			);
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while deleting the pattern.' );
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+				id: 'edit-site-patterns-error',
+			} );
+		}
+	};
+
+	const deleteItem = async () => {
+		if ( isTemplateRemovable( record ) ) {
+			removeTemplate( record );
+		} else if ( isUserPattern ) {
+			deletePattern( record );
+		}
+	};
+
+	async function revertAndSaveTemplate() {
+		try {
+			await revertTemplate( record, { allowUndo: false } );
+			await saveEditedEntityRecord( 'postType', record.type, record.id );
+
+			createSuccessNotice(
+				sprintf(
+					// translators: %s: the pattern, template/part's title.
+					__( '"%s" reverted.' ),
+					decodedTitle
+				),
+				{
+					type: 'snackbar',
+					id: 'edit-site-template-reverted',
+				}
+			);
+		} catch ( error ) {
+			const fallbackErrorMessage = sprintf(
+				// translators: %s: a post type label, e.g., Template, Template Part or Pattern.
+				__( 'An error occurred while reverting the %s.' ),
+				POST_TYPE_LABELS[ record?.type ] ??
+					POST_TYPE_LABELS[ TEMPLATE_POST_TYPE ]
+			);
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: fallbackErrorMessage;
+
+			createErrorNotice( errorMessage, { type: 'snackbar' } );
+		}
+	}
+
```

Comment:
It appears that an empty dropdown menu is rendered when there are no actions available.

![no-action](https://github.com/WordPress/gutenberg/assets/54422211/1929afcc-7f7c-4047-9616-70d448939f5c)

We may be able to come up with a better conditional statement, but if there is nothing that can be rendered, it seems better to return null.

```javascript
const showDropdown =
	isEditable ||
	isDuplicable ||
	isUserPattern ||
	isEditable ||
	isRevertable;

if ( ! showDropdown ) {
	return null;
}
```